src/sizzle.js

```diff
@@ -8,1150 +8,1198 @@
-					if ( b === a ) {
-						return true;
+		hasCompare = rnative.test( docElem.compareDocumentPosition );
+
+		// Element contains another
+		// Purposefully self-exclusive
+		// As in, an element does not contain itself
+		contains = hasCompare || rnative.test( docElem.contains ) ?
+			function( a, b ) {
+				var adown = a.nodeType === 9 ? a.documentElement : a,
+					bup = b && b.parentNode;
+				return a === bup || !!( bup && bup.nodeType === 1 && (
+					adown.contains ?
+						adown.contains( bup ) :
+						a.compareDocumentPosition && a.compareDocumentPosition( bup ) & 16
+				) );
+			} :
+			function( a, b ) {
+				if ( b ) {
+					while ( ( b = b.parentNode ) ) {
+						if ( b === a ) {
+							return true;
+						}
 					}
 				}
-			}
-			return false;
-		};
+				return false;
+			};
 
-	/* Sorting
+		/* Sorting
 	---------------------------------------------------------------------- */
 
-	// Document order sorting
-	sortOrder = hasCompare ?
-	function( a, b ) {
+		// Document order sorting
+		sortOrder = hasCompare ?
+			function( a, b ) {
 
-		// Flag for duplicate removal
-		if ( a === b ) {
-			hasDuplicate = true;
-			return 0;
-		}
+				// Flag for duplicate removal
+				if ( a === b ) {
+					hasDuplicate = true;
+					return 0;
+				}
 
-		// Sort on method existence if only one input has compareDocumentPosition
-		var compare = !a.compareDocumentPosition - !b.compareDocumentPosition;
-		if ( compare ) {
-			return compare;
-		}
+				// Sort on method existence if only one input has compareDocumentPosition
+				var compare = !a.compareDocumentPosition - !b.compareDocumentPosition;
+				if ( compare ) {
+					return compare;
+				}
 
-		// Calculate position if both inputs belong to the same document
-		compare = ( a.ownerDocument || a ) === ( b.ownerDocument || b ) ?
-			a.compareDocumentPosition( b ) :
+				// Calculate position if both inputs belong to the same document
+				compare = ( a.ownerDocument || a ) === ( b.ownerDocument || b ) ?
+					a.compareDocumentPosition( b ) :
 
-			// Otherwise we know they are disconnected
-			1;
+					// Otherwise we know they are disconnected
+					1;
 
-		// Disconnected nodes
-		if ( compare & 1 ||
-			(!support.sortDetached && b.compareDocumentPosition( a ) === compare) ) {
+				// Disconnected nodes
+				if ( compare & 1 ||
+					( !support.sortDetached && b.compareDocumentPosition( a ) === compare ) ) {
 
-			// Choose the first element that is related to our preferred document
-			if ( a === document || a.ownerDocument === preferredDoc && contains(preferredDoc, a) ) {
-				return -1;
-			}
-			if ( b === document || b.ownerDocument === preferredDoc && contains(preferredDoc, b) ) {
-				return 1;
-			}
+					// Choose the first element that is related to our preferred document
+					if ( a === document || a.ownerDocument === preferredDoc &&
+						contains( preferredDoc, a ) ) {
+						return -1;
+					}
+					if ( b === document || b.ownerDocument === preferredDoc &&
```

Comment:
Same remark: I'd prefer breaking the line after `||` as `&&` has precendence so it'd be simpler to understand.